docs/abi.md

```diff
@@ -91,7 +91,8 @@ functions** as
   for data (args 1 and 2) or handles (args 4 and 5) are not large enough for the
   read operation, then no data will be returned and either `BUFFER_TOO_SMALL` or
   `HANDLE_SPACE_TOO_SMALL` will be returned; in either case, the required sizes
-  will be returned in the spaces provided by args 3 and 6.
+  will be returned in the spaces provided by args 3 and 6. If no messages are
+  available on the channel, `CHANNEL_EMPTY` will be returned.
```

Comment:
I don't have a concrete proposal yet, but I am thinking about whether we could use the `read` and `write` host functions to operate on things other than channels (e.g. some sort of encapsulated message + policy, or other types of containers). Do you think we could keep this generic enough so that we could decide to use it for that, if we wanted? Basically drop the "Channel" part from the name, though I don't have a good suggestion for alternatives, apart from changing it completely to something like "must wait", similar to Fuchsia. Thoughts?

Reply:
TBH, it seems a bit premature to worry about naming one constant against the potential for some future re-design.  If we end up genericizing (is that even a word?) things to deal with more than just channels, renaming one extra constant will be a drop in the ocean &ndash; the "channel" name and concept is everywhere. 

(Also, I found "must wait" confusing as a name.)

Reply:
Fair enough, you are right that the host function names are already quite coupled to the channel terminology anyways.